### ISSUE TITLE
Exibição das mensagens de retorno

### ISSUE THREAD

**Issue by cevmelo** [servicosgovbr/editor-de-servicos#420]
O tempo de exibição das mensagens de retorno são insuficientes. Sugere-se que as mensagens sejam permanentes e possam ser fechadas clicando em um icone.

Ex: “Serviço ainda contém erros.”

**Comment by cevmelo** [comment]
@gustavo-freitas @sahrosa como podemos melhorar esse ponto

**Comment by sahrosa** [comment]
Oi @cevmelo, acho que podemos pensar em duas possibilidades:

1. Aumentar o tempo de permanência da mensagem.
2. Deixar a mensagem permanente até o usuário fechar. 

Sobre a opção 2, @glauberramos, sabe nos dizer se é muito complexo modificar do modelo atual (mensagem permanece por alguns segundos) por um modelo pertinente (onde o usuário precisa fechar a mensagem)?

Acredito que a opção 1 seja melhor, @cevmelo, pois como estas mensagens aparecerão de forma bastante repetitiva na interface, ficar pedindo para o usuário fecha-las todas as vezes vai ser muito cansativo e repetitivo. O que acha @glauberramos @gustavo-freitas ?

**Comment by glauberramos** [comment]
Tanto a 1 quanto a 2 não são difíceis de fazer. Acredito que o primeiro caso seja melhor porque vai evitar o usuário ter que fechar várias mensagens que se acumulam.

**Comment by gustavo-freitas** [comment]
Concordo @glauberramos. A primeira opção também parece ser mais rápida de alterar.

**Comment by cevmelo** [comment]
Não como deixar a mensagem estática com a opção de fechar e também fecha-la automaticamente caso o usuário execute uma nova ação que desencadeie a exibição de uma nova mensagem ?

Com isso apenas uma mensagem ficaria visível.

@sahrosa  @gustavo-freitas @glauberramos

**Comment by sahrosa** [comment]
Olá @cevmelo, acredito que misturar as duas opções vai ser mais complicado, pois temos algumas limitações em função do framework utilizado.

Podemos seguir com a opção 1?

**Comment by gustavo-freitas** [comment]
Na minha percepção, a opção 1 parece mais coerente para o nosso contexto!

2015-11-11 18:17 GMT-02:00 Samantha Rosa <notifications@github.com>:

**Comment by cevmelo** [comment]
Ok, prossiga com a implementação da opção 1.

**Comment by sahrosa** [comment]
Combinado @cevmelo :)

**Comment by glauberramos** [comment]
Coloquei o tempo de 4 segundos para as mensagens sumirem

**Comment by cevmelo** [comment]
@glauberramos a mensagem pode ser exibida por uns 5 minutos ?

**Comment by cevmelo** [comment]
@glauberramos @sahrosa  @gustavo-freitas O ideal é que a mensagem fique estática e o usuário tenha a possibilidade de fechar a mensagem. Além disso, um clique nos botões de ação (salvar, visualizar, publicar etc) também deve fechar a mensagem exibida.

Caso não seja possível, conversando com o Gustavo, chegamos no tempo de 30 segundos para a mensagem.

**Comment by gustavo-freitas** [comment]
@cevmelo vamos confirmar se 30 segundos é, de fato, um tempo ideal. A @sahrosa e o @glauberramos podem nos responder melhor essa questão.

**Comment by sahrosa** [comment]
Vamos fazer uns testes sim, @gustavo-freitas @cevmelo. 

@glauberramos podemos colocar 30s pra testar como fica?

**Comment by glauberramos** [comment]
Testamos com 30, 20, 15 e 10 segundos para as mensagens. O melhor tempo foi de 10 segundos.

Acreditamos que 10 segundos seja o ideal para o usuário entender o que está ocorrendo no editor e para poder tomar uma ação a partir disso.

Quando a mensagem fica muito tempo na tela ocorre acumulo de mensagens e gera poluição visual para o usuário, como pode ser visto na imagem a seguir:

<img width="907" alt="screen shot 2015-11-13 at 3 47 24 pm" src="https://cloud.githubusercontent.com/assets/559434/11153187/4e24ef68-8a1e-11e5-9dbd-04cc7e04d84c.png">

**Comment by sahrosa** [comment]
Oi @glauberramos @cevmelo @gustavo-freitas, acho que 10s ainda está longo em determinados casos, mas acho que é uma quantidade de tempo adequada.

Vamos fechar assim? :)

**Comment by gustavo-freitas** [comment]
Fechamos assim, Sá. Acredito que os testes que o @glauberramos
<https://github.com/glauberramos> fez deixaram claro que 10 segundos é um
tempo apropriado.

2015-11-13 16:07 GMT-02:00 Samantha Rosa <notifications@github.com>:

**Comment by almeidafab** [comment]
@gustavo-freitas Fizemos uma reunião ontem com a SEGEP e eles reclamaram bastante com a relação a mensagem sumir após 10s. Eles sugeriram deixar a cargo do usuário o descarte da mensagem (ex.: um botão para fechar a mensagem). Podemos conversar sobre isso?

**Comment by gustavo-freitas** [comment]
@almeidafab a possibilidade de fechar a mensagem já existe, só clicar na mensagem que ela será automaticamente fechada. Sobre a questão do tempo, o tempo que estamos utilizando é relativamente grande (10 segundos). Conforme conversas anteriores, isso dificulta o uso da aplicação do ponto de vista do usuário. Caso a mensagem venha a ficar estática, sem fechamento automático, isso vai ser ainda mais danoso para a experiência do usuário, forçando o mesmo a fechar cada umas das mensagens que a aplicação exibe.

**Comment by gustavo-freitas** [comment]
@sahrosa @glauberramos de acordo com conversas com @almeidafab e @mformiga a possibilidade de deixar a mensagem estática e deixar para o usuário fechar a mensagem é a melhor opção. Opiniões ou sugestões?

**Comment by glauberramos** [comment]
Acho que podemos deixar com tempo indefinido e colocar um X para mostrar que é possível fechar

**Comment by almeidafab** [comment]
boa @glauberramos

**Comment by glauberramos** [comment]
@almeidafab Ajustes feitos, pode conferir?

**Comment by mformiga** [comment]
@glauberramos conferido.

**Comment by sahrosa** [comment]
@glauberramos o botão de fechar está desalinhado. Temos como alinhar pelo centro?

<img width="757" alt="screen shot 2015-11-20 at 1 12 09 pm" src="https://cloud.githubusercontent.com/assets/2932644/11303120/6236a9dc-8f88-11e5-9c39-1ba9e2c5e477.png">

**Comment by almeidafab** [comment]
homologado. Passando para produção.
